src/solver/yaml_helper.jl

```diff
@@ -1,24 +1,149 @@
-yaml file, `yaml_file`. `filter_name` can be used to filter out
-required substrings from each elements of the vector.
-"""
-function commands_from_yaml(yaml_file; filter_name = nothing)
-    data = YAML.load_file(yaml_file)
-    steps = filter(x -> x isa Dict && haskey(x, "steps"), data["steps"])
-    cmds = map(x -> x["steps"], steps)
-    cmds = vcat(cmds...)
-    cmds = map(x -> x["command"], cmds)
-    cmds = vcat(cmds...)
-    cmds = map(x -> split(x, "\n"), cmds) # handle commands that run multiple scripts
-    cmds = vcat(cmds...)
-    cmds = map(x -> x isa AbstractString ? x : join(x), cmds)
-    if filter_name ≠ nothing
-        filter!(x -> occursin(filter_name, x), cmds)
+    default_config_dict()
+
+Loads the default configuration from files into a Dict for use in AtmosConfig().
+"""
+function default_config_dict()
+    default_config_file =
+        joinpath(pkgdir(CA), ".buildkite", "default_config.yml")
+    config = YAML.load_file(default_config_file)
+    edmf_config_file =
+        joinpath(pkgdir(CA), ".buildkite", "default_edmf_config.yml")
+    edmf_config = YAML.load_file(edmf_config_file)
+    # Combine base config with EDMF config - Don't allow duplicate entries.
+    isempty(intersect(keys(config), keys(edmf_config))) ||
+        error("Duplicate keys in default config and EDMF config.")
+    merge!(config, edmf_config)
+    # Strip out help messages
+    for (k, v) in config
+        config[k] = v["value"]
+    end
+    return config
+end
+
+"""
+    config_from_target_job(target_job)
+
+Given a job id string, returns the configuration for that job.
+Does not include the default configuration dictionary.
+"""
+function config_from_target_job(target_job)
+    for entry in configs_per_job_id(joinpath(pkgdir(CA), ".buildkite"))
+        if entry["job_id"] == target_job
+            return entry
+        end
+    end
+end
+
+"""
+    override_default_config(override_configs)
+
+Takes in a vector of dicts or filepaths and returns a dict with the
+default configuration overridden by the given dicts or parsed YAML files.
+"""
+function override_default_config(config_files::Vector{String})
+    override_default_config(YAML.load_file.(config_files))
+end
+
+function override_default_config(override_configs::Vector{Dict{Any, Any}})
+    default_config = default_config_dict()
+    config = deepcopy(default_config)
+    for override_config in override_configs
+        for (k, v) in override_config
+            !haskey(default_config, k) &&
+                error("Key `$k` is not in the default configuration.")
+            default_type = typeof(default_config[k])
+            config[k] = isnothing(default_config[k]) ? v : default_type(v)
+        end
+    end
+    return config
+end
+
+
+"""
+    non_default_config_entries(config)
+
+Given a configuration Dict, returns a Dict of the non-default values.
+"""
+function non_default_config_entries(config)
+    defaults = default_config_dict()
```

Comment:
```suggestion
function non_default_config_entries(config, defaults = default_config_dict())
```
I think this just tends to be more flexible